caddyconfig/httpcaddyfile/directives.go

```diff
@@ -398,6 +398,21 @@ func sortRoutes(routes []ConfigValue) {
 		if len(jPM) > 0 {
 			jPathLen = len(jPM[0])
 		}
+
+		// if both directives have no path matcher, use whichever one
+		// has any kind of matcher defined first.
+		if iPathLen == 0 && jPathLen == 0 {
+			return len(iRoute.MatcherSetsRaw) > len(jRoute.MatcherSetsRaw)
+		}
```

Comment:
Technically, this will sort by which one has more matcher sets than the other, but we don't know that that's the right order. What you want here instead is a return value conditional on whether the `i` or `j` route has 0 matchers and the other one has more than 0.

Reply:
True, but I don't think it's ever possible for a Caddyfile directive to have more than one matcher set. But it's an easy fix.